server/adka2a/executor_context.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adka2a
+
+import (
+	"context"
+
+	"github.com/a2aproject/a2a-go/a2asrv"
+
+	"google.golang.org/genai"
+
+	"google.golang.org/adk/session"
+)
+
+type ExecutorContext interface {
+	context.Context
+
+	SessionID() string
+	UserID() string
+	AgentName() string
+	ReadonlyState() session.ReadonlyState
+	UserContent() *genai.Content
+	RequestContext() *a2asrv.RequestContext
+}
+
+type executorContext struct {
+	context.Context
+	meta        invocationMeta
+	session     session.ReadonlyState
+	userContent *genai.Content
+}
+
+var _ ExecutorContext = (*executorContext)(nil)
```

Comment:
nit: up to your preference, but this line can be removed, because this check is already in lines 47-48